pkg/k8s/k8s_replicaset.go

```diff
@@ -12,62 +12,36 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// getReplicaSet retrieves replicaSet from the given namespace and logs any errors.
-func getReplicaSet(ctx context.Context, namespace, name string) (*appv1.ReplicaSet, error) {
-	if !IsInitialized() {
-		return nil, ErrKnuuNotInitialized
-	}
-
-	replicaset, err := Clientset().AppsV1().ReplicaSets(namespace).Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		return nil, ErrGettingReplicaSet.WithParams(name).Wrap(err)
-	}
-
-	return replicaset, nil
+type ReplicaSetConfig struct {
+	Name      string            // Name of the ReplicaSet
+	Namespace string            // Namespace of the ReplicaSet
+	Labels    map[string]string // Labels to apply to the ReplicaSet, key/value represents the name/value of the label
+	Replicas  int32             // Replicas is the number of replicas
+	PodConfig PodConfig         // PodConfig represents the pod configuration
 }
 
-// DeployReplicaSet creates a new replicaSet in the given namespace if it doesn't already exist.
-func DeployReplicaSet(ctx context.Context, replicaSetConfig ReplicaSetConfig, init bool) (*appv1.ReplicaSet, error) {
+// DeployReplicaSet creates a new replicaSet in namespace that k8s is initialized with if it doesn't already exist.
+func (c *Client) DeployReplicaSet(ctx context.Context, rsConfig ReplicaSetConfig, init bool) (*appv1.ReplicaSet, error) {
```

Comment:
We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?